### PR TITLE
Fix issues

### PR DESCRIPTION
**Pull request by charithag**

### CODE REVIEW

File: modules/samples/doormanager/component/controller/src/main/java/org.homeautomation/doormanager/controller/api/DoorManagerControllerService.java [hunk 1]
```diff
@@ -76,6 +77,8 @@
 
     private PrivilegedCarbonContext ctx;
 
+    @Context  //injected response proxy supporting multiple thread
+
```
Comment (DilanUA) [inline modules/samples/doormanager/component/controller/src/main/java/org.homeautomation/doormanager/controller/api/DoorManagerControllerService.java:81]: unnecessary new line.

File: modules/samples/doormanager/component/controller/src/main/java/org.homeautomation/doormanager/controller/api/DoorManagerControllerService.java [hunk 2]
```diff
@@ -144,36 +147,36 @@ public void assignUseToLock(@HeaderParam("owner") String owner,
...
                     }
+                } catch (UserStoreException e) {
+                    response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+                    log.error(e);
                 }
-                else {
-                    response.setStatus(Response.Status.BAD_REQUEST.getStatusCode());
-                }
+            } else {
+                response.setStatus(Response.Status.BAD_REQUEST.getStatusCode());
```
Comment (DilanUA) [inline modules/samples/doormanager/component/controller/src/main/java/org.homeautomation/doormanager/controller/api/DoorManagerControllerService.java:178]: better to log the issue as well here.

Comment (charithag) [inline modules/samples/doormanager/component/controller/src/main/java/org.homeautomation/doormanager/controller/api/DoorManagerControllerService.java:178]: This only applicable for bad request. Why we should log bad request from client side in server side as an error?

File: modules/samples/doormanager/component/controller/src/main/java/org.homeautomation/doormanager/controller/api/DoorManagerControllerService.java (hunk 1 above)
Comment (charithag) [inline modules/samples/doormanager/component/controller/src/main/java/org.homeautomation/doormanager/controller/api/DoorManagerControllerService.java:81]: Fixed in commit https://github.com/wso2/product-iots/commit/1f06c08f0d87e3161e7a135e9d4496e757460247

**Pull request merged**